build.rs

```diff
@@ -5,7 +5,34 @@ use std::path::Path;
 
 fn main() -> io::Result<()> {
     let out_dir = env::var_os("OUT_DIR").unwrap();
-    let target = env::var("TARGET").ok();
+    let mut target = env::var("TARGET").ok();
+
+    // When --target flag is passed, cargo does not pass RUSTFLAGS to rustc when
+    // building proc-macro and build script even if the host and target triples
+    // are the same. Therefore, if we always pass --target to cargo, tools such
+    // as coverage that require RUSTFLAGS do not work for tests run by trybuild.
+    //
+    // To avoid that problem, do not pass --target to cargo if we know that it
+    // has not been passed.
+    //
+    // Cargo does not have a way to tell the build script whether --target has
+    // been passed or not, so we use the following heuristic:
+    //
+    // - The host and target triples are the same.
+    // - And RUSTFLAGS is available when *building* the build script.
```

Comment:
I think this shouldn't be a problem because the build script will be rebuilt when RUSTFLAGS is changed, but I may have missed something.